test/Concurrency/Runtime/async.swift

```diff
@@ -38,7 +37,7 @@ if #available(SwiftStdlib 5.5, *) {
   asyncTests.test("Detach") {
     detach(priority: .background) {
       async {
-        assert(Task.currentPriority == .background)
+        assert(Task.currentPriority == .default)
```

Comment:
1. Should `currentPriority` be returning `TaskPriority?` instead of `TaskPriority` ?
2. Should we be using `medium` here instead of `default`? (And similarly update `currentPriority`'s doc comment.)

Reply:
`currentPriority` queries the running thread if there's no active task